src/streaming/controllers/EventController.js

```diff
@@ -537,6 +557,32 @@ function EventController() {
         }
     }
 
+    /**
+     * Patch the manifest
+     * @private
+     */
+    function _patchManifest(manifestPatch) {
+        try {
+            checkConfig();
+            manifestUpdater.patchManifest(manifestPatch);
+        } catch (e) {
+            logger.error(e);
+        }
+    }
+
+    /**
+     * Update the manifest
+     * @private
+     */
+    function _updateManifest(manifest) {
+        try {
+            checkConfig();
+            manifestUpdater.updateManifest(manifest);
+        } catch (e) {
+            logger.error(e);
+        }
```

Comment:
I think instead of doing this we should only trigger the INTERNAL_MANIFEST_LOAD event
```suggestion
        try {
            checkConfig();
            // ALSO DO PARSE STUFF HERE
            eventBus.on(Events.INTERNAL_MANIFEST_LOADED, { manifest });
        } catch (e) {
            logger.error(e);
        }
```